### PR TITLE
TST: Add tests for page labels

### PR DESCRIPTION
**Pull request by pubpub-zz**

### CODE REVIEW

**Comment by pubpub-zz** [comment]
from https://www.furorteutonicus.eu/2015/01/12/pdf-page-labels-test/?upm_export=text
I found this PDF for testing PageLabels:
[waarom-meisjes-het-beter-doen-op-HAVO-en-VWO-ROA.pdf](https://github.com/py-pdf/pypdf/files/10773829/waarom-meisjes-het-beter-doen-op-HAVO-en-VWO-ROA.pdf)

**Comment by pubpub-zz** [comment]
@MartinThoma
all yours

File: pypdf/_page_labels.py [hunk 1]
```diff
@@ -159,8 +159,7 @@ def index2label(reader: PdfReaderProtocol, index: int) -> str:
             "/A": number2uppercase_letter,
             "/a": number2lowercase_letter,
         }
-        if not isinstance(value, dict):
-            value = reader.get_object(value)
+        # if Nums array not iaw standards and if Nums is empty
```
Comment (MartinThoma) [inline pypdf/_page_labels.py:162]: I don't understand that comment. What does "iaw" mean?

File: tests/test_page_labels.py [hunk 2]
```diff
@@ -44,3 +60,45 @@ def test_number2lowercase_roman_numeral():
...
 def test_number2lowercase_letter(number, expected):
     assert number2lowercase_letter(number) == expected
+
+
+def test_number2uppercase_letter():
+    with pytest.raises(ValueError):
+        number2uppercase_letter(-1)
+
+
+def test_index2label(caplog):
```
Comment (MartinThoma) [inline tests/test_page_labels.py:70]:
```suggestion
@pytest.mark.external
def test_index2label(caplog):
```

File: pypdf/_page_labels.py (hunk 1 above)
Comment (pubpub-zz) [reply pypdf/_page_labels.py:162]: "In Accordance With" (https://en.wiktionary.org/wiki/IAW) 
I use it often but less than ASAP 😂

File: tests/test_page_labels.py (hunk 2 above)
Comment (pubpub-zz) [reply tests/test_page_labels.py:70]: oups thanks

File: pypdf/_page_labels.py (hunk 1 above)
Comment (MartinThoma) [inline pypdf/_page_labels.py:162]:
```suggestion
        # if /Nums array is not following the specification or if /Nums is empty
```

File: pypdf/_page_labels.py [hunk 3]
```diff
@@ -176,7 +175,7 @@ def index2label(reader: PdfReaderProtocol, index: int) -> str:
             __name__,
         )
     # TODO: Implement /Kids and /Limits for number tree
-    return str(index + 1)  # Fallback
+    return str(index + 1)  # Fallback (if Kids or no Nums
```
Comment (MartinThoma) [inline pypdf/_page_labels.py:178]:
```suggestion
    return str(index + 1)  # Fallback if /Nums is not in the number_tree
```

**Pull request merged**